src/main/java/target/HttpTarget.java

```diff
@@ -111,14 +112,23 @@ public CompletableFuture<Object> callBatch(
                 .create(batchRequestId, targetRequestId)
                 .compose(client.newCall(request))
                 .executeAsync()
-                .handleAsync((response, throwable) ->
-                    Integer
-                            .toString(response.code())
-                            .matches(Config.PRODUCE_TO_DEAD_LETTER_TOPIC_WHEN_STATUS_CODE_MATCH) ||
-                        throwable != null
-                        ? new TargetException(response, throwable)
-                        : null
-                );
+                .handleAsync((response, throwable) -> {
+                    try {
+                        if (
+                            Integer
+                                .toString(response.code())
+                                .matches(Config.PRODUCE_TO_DEAD_LETTER_TOPIC_WHEN_STATUS_CODE_MATCH) ||
+                            throwable != null
+                        ) {
+                            assert response.body() != null;
+                            return new TargetException(response.code(), response.body().string(), throwable);
```

Comment:
return an exception?

Reply:
yes, its not thrown but used in the upper Observable.
Its not new in this PR